spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskDefinitionController.java

```diff
@@ -163,15 +163,23 @@ public void destroyAll() {
 	@RequestMapping(value = "", method = RequestMethod.GET)
 	@ResponseStatus(HttpStatus.OK)
 	public PagedModel<TaskDefinitionResource> list(Pageable pageable, @RequestParam(required = false) String search,
-			@RequestParam(required = false) boolean manifest,
+			@RequestParam(required = false) boolean manifest, @RequestParam(required = false) String dslText,
 			PagedResourcesAssembler<TaskExecutionAwareTaskDefinition> assembler) {
 
 		final Page<TaskDefinition> taskDefinitions;
 		if (search != null) {
-			taskDefinitions = repository.findByTaskNameContains(search, pageable);
+			if (dslText != null) {
+				taskDefinitions = repository.findByTaskNameContainsAndDslTextContains(search, dslText, pageable);
```

Comment:
Not sure this query is required.  Because task definition names are unique in dataflow.   
In this case we may want to throw a `TaskQueryParamException` (a new type of exception) stating that you can't request dsltext and search at the same time.   Then add the new `TaskQueryParamException` to the `RestControllerAdvice` here: https://github.com/spring-cloud/spring-cloud-dataflow/blob/master/spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/RestControllerAdvice.java#L180-L187

Reply:
@cppwfs so you are saying that we don't want task name and dsl to be searched together? Should I just create a separate API for searching by dslText in that case?

Reply:
@cppwfs So you are saying that we don't want to merge the filters for task definition list? Allow user to only have one filter? i.e. either task name, dslText, description or status, but no combination ?

Reply:
I have made the requested changes, can you please check again? Thanks.